os/fs/smartfs/smartfs_utils.c

```diff
@@ -1235,9 +1236,9 @@ int smartfs_finddirentry(struct smartfs_mountpt_s *fs, struct smartfs_entry_s *d
 									/* Read the next sector of the file */
 
 									smartfs_setbuffer(&readwrite, dirsector, 0, sizeof(struct smartfs_chain_header_s), (uint8_t *)fs->fs_rwbuffer);
-									ret = FS_IOCTL(fs, BIOC_READSECT, (unsigned long)&readwrite);
-									if (ret < 0) {
-										fdbg("Error in sector chain at %d, ret : %d\n", readwrite.logsector, ret);
+									tmp = FS_IOCTL(fs, BIOC_READSECT, (unsigned long)&readwrite);
+									if (tmp < 0) {
```

Comment:
instead of tmp, can we use ret only and assign ret = o or -ENOENT in this case

Reply:
Updated for return values in case IOCTL fails, please review